### ISSUE TITLE
Worker Thread for AsyncCaller

### ISSUE THREAD

**Issue by Rincewind34** [hogliux/farbot#11]
I just took a look into this library (and watched the two videos on YouTube) but I can't find a good example on how to run the process function. In the video you said that it is possible to make the process thread run interleaved with audio thread? How exactly can I accomplish that?
Can you recommend any papers I can read about that topic? When I search on that topics, I tend to find a lot of stuff about writing the audio thread and the syncing mechanisms (like a fifo queue) but its very hard to find example for the consumer thread that is actually working the queue. Even though that may only be very few lines of code they are very important.

I hope you (or someone else of course) can help me with that.
Thanks in advance!

**Comment by tomjwaldron** [comment]
There's a partial example here: https://hogliux.github.io/farbot/presentations/meetingcpp_2019/assets/player/KeynoteDHTMLPlayer.html#172

This just uses a juce::Timer callback to run the process function, the timer callback itself would be run from the juce message thread.

**Comment by hogliux** [comment]
Yes, you need to have some kind of idle or timer callback function which regularly checks if there is work to do.
